### PR TITLE
don't enforce maxuploadtarget's disconnect for whitelisted peers

### PR DESCRIPTION
**Pull request by jonasschnelli**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -3856,8 +3856,9 @@ void static ProcessGetData(CNode* pfrom, const Consensus::Params& consensusParam
                     }
                 }
                 // disconnect node in case we have reached the outbound limit for serving historical blocks
+                // never disconnect whitelisted nodes
                 static const int nOneWeek = 7 * 24 * 60 * 60; // assume > 1 week = historical
-                if (send && CNode::OutboundTargetReached(true) && ( ((pindexBestHeader != NULL) && (pindexBestHeader->GetBlockTime() - mi->second->GetBlockTime() > nOneWeek)) || inv.type == MSG_FILTERED_BLOCK) )
+                if (send && CNode::OutboundTargetReached(true) && ( ((pindexBestHeader != NULL) && (pindexBestHeader->GetBlockTime() - mi->second->GetBlockTime() > nOneWeek)) || inv.type == MSG_FILTERED_BLOCK) && !pfrom->fWhitelisted )
```
Comment (sipa) [inline src/main.cpp]: Unnecessary space before )

**Comment by sipa** [comment]
utACK

File: src/main.cpp (hunk 1 above)
Comment (jonasschnelli) [inline src/main.cpp]: True. Force-push removed.

**Comment by gmaxwell** [comment]
utACK

**Comment by fanquake** [comment]
utACK

**Comment by laanwj** [comment]
Concept ACK

**Comment by instagibbs** [comment]
utACK

File: qa/rpc-tests/maxuploadtarget.py [hunk 2]
```diff
@@ -244,5 +244,38 @@ def run_test(self):
...
+
+        for i in xrange(3):
+            test_nodes.append(TestNode())
+            connections.append(NodeConn('127.0.0.1', p2p_port(0), self.nodes[0], test_nodes[i]))
+            test_nodes[i].add_connection(connections[i])
+
+        NetworkThread().start() # Start up network handling in another thread
+        [x.wait_for_verack() for x in test_nodes]
+
+        #retrieve 20 blocks which should be enought to break the 1MB limit
```
Comment (sipa) [inline qa/rpc-tests/maxuploadtarget.py]: nit: enough

**Comment by sipa** [comment]
utACK

File: qa/rpc-tests/maxuploadtarget.py (hunk 2 above)
Comment (jonasschnelli) [inline qa/rpc-tests/maxuploadtarget.py]: Thanks. Fixed.

File: src/main.cpp [hunk 3]
```diff
@@ -3856,8 +3856,9 @@ void static ProcessGetData(CNode* pfrom, const Consensus::Params& consensusParam
                     }
                 }
                 // disconnect node in case we have reached the outbound limit for serving historical blocks
+                // never disconnect whitelisted nodes
```
Comment (MarcoFalke) [inline src/main.cpp]: Is this already mentioned in the release notes and/or documentation?

Comment (gmaxwell) [inline src/main.cpp]: I think it's more or less adequately explained by the documentation for whitelist.

Though the release note for maxupload target could mention it too.

**Comment by gmaxwell** [comment]
Also tested, works good. (I gave a utACK above).   Please update docs/release notes as you see fit.

**Comment by jonasschnelli** [comment]
I have added a short documentation in the `release-notes.md` and in the `reduce-traffic.md` doc (feel free to overhaul). Also i smuggled in a rename from `reducetraffic.md` to `reduce-traffic.md` (which would be silly to do in a separated PR). All other doc filename use the same pattern.

**Comment by jonasschnelli** [comment]
Thanks. Fixed.

**Pull request merged**